FORMAT.md

```diff
@@ -0,0 +1,226 @@
+     +-------+-------+-------+-------+
+```
+
+#### Int64 (`0x3`)
+```text
+Byte:  0       1       2       3       4       5       6       7
+     +-------+-------+-------+-------+-------+-------+-------+-------+
+     | LSB                                                     MSB   |
+     | (Little-endian i64)                                           |
+     +-------+-------+-------+-------+-------+-------+-------+-------+
+```
+
+#### Float32 (`0x4`)
+```text
+Byte:  0       1       2       3
+     +-------+-------+-------+-------+
+     | (IEEE 754 32-bit float)       |
+     | (Little-endian)               |
+     +-------+-------+-------+-------+
+```
+
+#### Float64 (`0x5`)
+```text
+Byte:  0       1       2       3       4       5       6       7
+     +-------+-------+-------+-------+-------+-------+-------+-------+
+     | (IEEE 754 64-bit float)                                       |
+     | (Little-endian)                                               |
+     +-------+-------+-------+-------+-------+-------+-------+-------+
+```
+
+### Variable-Width Types
+
+#### Bytes (`0x6`)
+```text
+     +---------------------+---------------------------------------+
+     | Length (varint)     | Raw Bytes Content                     |
+     | (1-5 bytes)         | (Length bytes)                        |
+     +---------------------+---------------------------------------+
+```
+
+#### String (`0x7`)
+```text
+     +---------------------+---------------------------------------+
+     | Length (varint)     | UTF-8 Encoded String                  |
+     | (1-5 bytes)         | (Length bytes)                        |
+     +---------------------+---------------------------------------+
+```
+
+#### Array (`0x8`)
+```text
+     +---------------------+-------+--------------------------------+
+     | Length (varint)     | Elem. | Element 1 | Element 2 | ...    |
+     | (1-5 bytes)         | Type  | (format depends on type)       |
+     +---------------------+-------+--------------------------------+
+                            ^
+                            Only present if Length > 0
+```
+
+#### Map (`0x9`)
+```text
+     +---------------------+-------+-------+------------------------+
+     | Length (varint)     | Key   | Value | Key 1   | Value 1 |... |
+     | (1-5 bytes)         | Type  | Type  | (format depends on respective types) |
+     +---------------------+-------+-------+------------------------+
+                            ^       ^       ^
+                            Only present if Length > 0
+```
+
+Valid map key types:
+- Int32 (`0x2`)
+- Int64 (`0x3`)
+- Bytes (`0x6`)
+- String (`0x7`)
+
+#### Row (`0xA`)
+```text
+     +----------------+-------------------------+---------------+
+     | Header (15B)   | Field Directory (opt.)  | Payload       |
+     +----------------+-------------------------+---------------+
+     (Complete Imprint record - recursive structure)
+```
+
+## Varint Encoding
```

Comment:
nice - this is actually one of the most concise explanations of varint I've seen!